### PR TITLE
refactor: Migrate ContentViewModel to Typescript

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/view_model/ContentViewModel.ts [hunk 1]
```diff
@@ -48,8 +50,85 @@ import {PreferencesDevicesViewModel} from './content/PreferencesDevicesViewModel
...
+import type {ServerTimeHandler} from '../time/serverTimeHandler';
+import type {StorageRepository} from '../storage';
+import type {TeamRepository} from '../team/TeamRepository';
+import type {BackupRepository} from '../backup/BackupRepository';
+import type {CallingRepository} from '../calling/CallingRepository';
+import type {UserRepository} from '../user/UserRepository';
+import type {Conversation} from '../entity/Conversation';
+import type {Message} from '../entity/message/Message';
+
+type Repositories = {
```
Comment (ffflorian) [inline src/script/view_model/ContentViewModel.ts]:
```suggestion
interface Repositories {
```

File: src/script/view_model/ContentViewModel.ts [hunk 2]
```diff
@@ -48,8 +50,85 @@ import {PreferencesDevicesViewModel} from './content/PreferencesDevicesViewModel
...
+  preferenceNotification: PreferenceNotificationRepository;
+  properties: PropertiesRepository;
+  search: SearchRepository;
+  serverTime: ServerTimeHandler;
+  storage: StorageRepository;
+  team: TeamRepository;
+  user: UserRepository;
+};
+
+type ShowConversationOptions = {
```
Comment (ffflorian) [inline src/script/view_model/ContentViewModel.ts]:
```suggestion
interface ShowConversationOptions {
```

File: src/script/view_model/ContentViewModel.ts [hunk 3]
```diff
@@ -48,8 +50,85 @@ import {PreferencesDevicesViewModel} from './content/PreferencesDevicesViewModel
...
+  exposeMessage: Message;
+  openFirstSelfMention: boolean;
+  openNotificationSettings: boolean;
```
Comment (ffflorian) [inline src/script/view_model/ContentViewModel.ts]:
```suggestion
  exposeMessage?: Message;
  openFirstSelfMention?: boolean;
  openNotificationSettings?: boolean;
```

File: src/script/view_model/ContentViewModel.ts [hunk 4]
```diff
@@ -228,19 +298,10 @@ export class ContentViewModel {
...
-   *
-   * @param {Conversation|string} conversation Conversation entity or conversation ID
-   * @param {Object} options State to open conversation in
-   * @param {Message} [options.exposeMessage] Scroll to message and highlight it
-   * @param {boolean} [options.openFirstSelfMention=false] Open first self mention instead of passed message
-   * @param {boolean} [options.openNotificationSettings=false] Open notification settings of conversation
-   * @returns {undefined} No return value
-   */
-  showConversation = (conversation, options = {}) => {
+  showConversation = (
```
Comment (ffflorian) [inline src/script/view_model/ContentViewModel.ts:312]: Please re-add the JSDoc without the type declarations. Example:

```js
/**
 * ...
 * @param conversation Conversation entity or conversation ID
 * ...
 */

File: src/script/view_model/ContentViewModel.ts [hunk 5]
```diff
@@ -401,17 +466,17 @@ export class ContentViewModel {
         this.conversationRepository.active_conversation(null);
       }
 
-      return this.messageList.release_conversation();
+      return this.messageList.release_conversation(undefined);
```
Comment (ffflorian) [inline src/script/view_model/ContentViewModel.ts:480]: Why did you add `undefined` here?

Comment (thisisamir98) [reply src/script/view_model/ContentViewModel.ts:480]: messageList.release_conversation excepts an argument but it's handling it if nothing is passed so typescript compiler was unhappy about this.

File: src/script/view_model/ContentViewModel.ts (hunk 4 above)
Comment (thisisamir98) [reply src/script/view_model/ContentViewModel.ts:312]: https://github.com/wireapp/wire-webapp/pull/9173/commits/4427041e2014c1861a0b8181a295ff1ffba0d50a
